src/firefly/js/ui/ImageSelect.jsx

```diff
@@ -291,7 +291,7 @@ function FilterPanelView({imageMasterData}) {
  * @param {object[]} p.dataList
  * @param {number}   p.maxShown
  */
-function FilterSelect ({type, dataList, maxShown=6}) {
+function FilterSelect ({type, dataList, maxShown=6, groupKey}) {
```

Comment:
I think you are working too hard passing `groupKey` you can just use context.
```js
const {groupKey}= useContext(FieldGroupCtx);
```

You can look at `FieldGroup.jsx`, `makeCtx()` (line 16) to see what is in the FieldGroupCtx